src/checkstyle/checks/size/MethodCountCheck.hx

```diff
@@ -0,0 +1,52 @@
+package checkstyle.checks.size;
+
+import checkstyle.token.TokenTree;
+import haxeparser.Data;
+
+using checkstyle.utils.ArrayUtils;
+
+@name("MethodCount")
+@desc("Checks the number of methods declared in each type. This includes the number of each scope (`private` and `public`) as well as an overall total.")
+class MethodCountCheck extends Check {
+
+	static var DEFAULT_MAX_COUNT:Int = 100;
+
+	public var maxTotal:Int;
+	public var maxPrivate:Int;
+	public var maxPublic:Int;
+
+	public function new() {
+		super(TOKEN);
+		maxTotal = DEFAULT_MAX_COUNT;
+		maxPrivate = DEFAULT_MAX_COUNT;
+		maxPublic = DEFAULT_MAX_COUNT;
+		categories = [Category.COMPLEXITY];
+		points = 21;
+	}
+
+	override function actualRun() {
+		var root:TokenTree = checker.getTokenTree();
+		var acceptableTokens:Array<TokenTree> = root.filter([Kwd(KwdFunction)], ALL);
```

Comment:
`ALL` walks through every node, giving you not only members, but also inner functions, so `FIRST` would be a better choice